specification/ingest/_types/Processors.ts

```diff
@@ -1046,6 +1187,42 @@ export class LowercaseProcessor extends ProcessorBase {
   target_field?: Field
 }
 
+export class NetworkDirectionProcessor extends ProcessorBase {
+  /**
+   * Field containing the source IP address.
+   * @server-default source.ip
+   */
+  source_ip?: Field
+  /**
+   * Field containing the destination IP address.
+   * @server-default destination.ip
+   */
+  destination_ip?: Field
+  /**
+   * Output field for the network direction.
+   * @server-default network.direction
+   */
+  target_field?: Field
+  /**
+   * List of internal networks. Supports IPv4 and IPv6 addresses and ranges in
+   * CIDR notation. Also supports the named ranges listed below. These may be
+   * constructed with template snippets. Must specify only one of
+   * internal_networks or internal_networks_field.
+   */
+  internal_networks: string[]
```

Comment:
This is the one field that is _almost_ an IP address where would could use the `Ip[]` type. But that would be inaccurate because the values can be a IP or a CIDR. Should I create a type alias for CIDR somewhere such that the type can be `(Ip | CIDR)[]`?

Reply:
> Please refresh my memory. Is xxx.xxx.xxx.xxx without a slash a valid CIDR notation that defaults to /32?

Yes, that's how the processor is behaving for ipv4. I didn't check it, but for ipv6 I would expect similar behavior just using /128 to indicate that it's matching one address.